### ISSUE TITLE
A valid json which not pass

### ISSUE THREAD

**Issue by frokaikan** [hjiang/jsonxx#71]
[JSONXX] expression 'read.size()' failed at jsonxx_test.cc:619 -> jsonxx_test: jsonxx.cc:29: void jsonxx::assertion(const char *, int, const char *, bool): Assertion `0' failed.
My json is as following:

{
    "string" : "lorem ipsum",
    "utf string" : "\u006corem\u0020ipsum",
    "utf-8 string": "あいうえお",
    "surrogate string": "lorem\uD834\uDD1Eipsum\uD834\uDF67lorem",
    "positive one" : 1,
    "negative one" : -1,
    "pi" : 3.14,
    "hard to parse number" : -3.14e-4,
    "big int": 2147483647,
    "big uint": 4294967295,
    "boolean true" : true,
    "boolean false" : false,
    "null" : null,
    "string array" : ["lorem", "ipsum"],
    "x^2 array" : [0, 1, 4, 9, 16, 25, 36, 49, 64, 81, 100],
    "/*" : null,
    "object" : { "nested string" : "str",
                 "nested true" : true,
                 "nested false" : false,
                 "nested null" : null,
                 "nested number" : 123,
                 "nested array" : ["lorem", "ipsum"],
                 "nested object" : {"lorem": "ipsum"}
    },
    "*/" : null,
    "/**/" : "comment",
    "//" : "comment",
    "url" : "https:\/\/www.example.com\/search?q=12345",
    "escaped chars" : "\" \\ \/",
    "empty object" : {},
    "empty array" : []
}

**Comment by lethosor** [comment]
Please wrap your JSON in triple backticks (```) so it's not formatted as markdown.

**Comment by frokaikan** [comment]
I don't know what you mean. You say I should change the json_file like the following?
```
{"integer" : 42}
```

I tried this but still can't read the file.

**Comment by lethosor** [comment]
No, do that in your comment *above*. I can't use the JSON you posted because your * characters were removed, and maybe other things changed too.

**Comment by frokaikan** [comment]
Ah , I found something wrong with my input file.
Sorry to bother you.
